pyspeckit/cubes/tests/test_cubetools.py

```diff
@@ -41,3 +92,99 @@ def test_subimage_integ_header():
 
     np.testing.assert_almost_equal(x1,x2)
     np.testing.assert_almost_equal(y1,y2)
+
+def do_fiteach(save_cube=None, save_pars=None, show_plot=False):
+    """
+    A draft version for running a simple test
+    on fiteach(). For now only the line widths
+    of gaussian fits are being checked.
+
+    Generates a fits file and checks that the
+    fitted line width is within expected values.
+    """
+    if save_cube is None:
+        save_cube = 'test.fits'
+    test_sigma = 10 # in pixel values, each pixel is CDELT3 thick
+    make_test_cube((100,10,10), save_cube, 
+                   sigma=(test_sigma, 5) )
+    # from .. import Cube
+    spc = Cube(save_cube)
+    guesses = [0.5,0.2,0.8]
+    map_rms = np.zeros_like(spc.cube[0])+spc.header['RMSLVL']
+    map_snr = spc.cube.std(axis=0)/map_rms
+    try:
+        import multiprocessing
+        ncores = multiprocessing.cpu_count()
+    except ImportError:
+        ncores = 1
+        pass
+    spc.fiteach(fittype = 'gaussian',
+	        guesses = guesses,
+	        start_from_pixel = (5,5),
+	    	multicore = ncores,
+                blank_value = np.nan,
+	    	verbose_level = 3,
+                errmap = map_rms,
+                signal_cut = 5)
+    if show_plot:
+        spc.mapplot()
+    if save_pars:
+        spc.write_fit(save_pars, clobber=True)
+    return spc
+
+def test_fiteach(save_cube=None, save_pars=None, show_plot=False):
+    """
+    A draft version for runing a simple test
+    on fiteach(). For now only the line widths
+    of gaussian fits are being checked.
+
+    Generates a fits file and checks that the
+    fitted line width is within expected values.
+    """
+    spc = do_fiteach(save_cube, save_pars, show_plot)
+
+    # checking the fit
+    map_sigma_post = spc.parcube[2]
+    map_sigma_true = np.zeros_like(map_sigma_post) + spc.header['SIGMA']
+    map_in_bounds = np.abs(map_sigma_true-map_sigma_post) < spc.errcube[2]
+    err_frac = map_in_bounds[~map_in_bounds].size / float(map_sigma_post.size)
+    # TODO: this is not an ideal aproach - should probably do proper statistics
+    #       on the expected value of median sigma value and check that it lies
+    #       within something like 99.7% of the expected sigma's PDF
+    # make sure the fraction of the fits that don't 
+    # include the true value is within tolerance
+    # TODO: rewrite - the condition is too strict!
+    # if 1-.68 is a peak of the expected value for
+    # the error fraction then this will fail half
+    # of the time. Letting it pass for now.
+    try:
+        assert err_frac < 1-.68, "the fraction of true values outside "+\
+                                 "the confidence intervals is too high "+\
+                                 "({}%)".format(err_frac*100)
```

Comment:
Instead of this "rigorous" use of a 1-sigma error bar, just look at the map and make sure the fits look good on some fraction.  Then, determine - given this specific realization of the noise - how many fits should be "good" by some *arbitrary* criterion and test that that value remains constant.